habitat/config/default.py

```diff
@@ -6,7 +6,15 @@
 
 from typing import List, Optional, Union
 
-from habitat.config import Config as CN  # type: ignore
+import yacs.config
+
+# from habitat.config import Config as CN # type: ignore
+
+# Default config node
+Config = lambda *args, **kwargs: yacs.config.CfgNode(
```

Comment:
We should do changing the default value of `new_allowed` via a subclass.  That way static analysis, code completion, and mypy will still work correctly.